pkg/internal/ebpf/common/common.go

```diff
@@ -231,10 +231,21 @@ func cstr(chars []uint8) string {
 }
 
 func (connInfo *BPFConnInfo) reqHostInfo() (source, target string) {
+	oneAddress := [16]uint8{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1}
```

Comment:
This is technically the IPv6 Loopback address. I'm still concerned that we have an event with it, but I don't mind putting a check to verify it's valid. 

The Go library code does it like this:
```
func (ip IP) IsLoopback() bool {
	if ip4 := ip.To4(); ip4 != nil {
		return ip4[0] == 127
	}
	return ip.Equal(IPv6loopback)
}
```

So I think all we need to do is:

```
if src.Equal(net.IPv6loopback) {
  src = ""
}

if dst.Equal(net.IPv6loopback) {
  dst = ""
}

```

Reply:
that is much cleaner, updated ....

on another note, does it make sense to report ipv4 loopback addresses in traces?

Reply:
I'd say it might have sense. E.g. if you run Beyla on a baremetal host that runs multiple services.

You can always use filters to remove local traces: https://grafana.com/docs/beyla/latest/configure/options/#filter-metrics-and-traces-by-attribute-values